From a68b0e27b6c60921c97f61daa2d9d723e0a5d98f Mon Sep 17 00:00:00 2001
From: Paulo Ricardo Zanoni <pzanoni@mandriva.com>
Date: Thu, 11 Mar 2010 14:28:18 -0300
Subject: [PATCH 02/25] dix: be more verbose when we run out of opcodes

If we run out of opcodes, nothing is print on the log, making the
problem hard to debug. In the current Xserver, if you enable some
extensions like multibuffer (+2 events) and use nvidia binary driver (+5
events) you can run out of opcode numbers.

Signed-off-by: Paulo Ricardo Zanoni <pzanoni@mandriva.com>
Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Julien Cristau <jcristau@debian.org>
Signed-off-by: Keith Packard <keithp@keithp.com>
(cherry picked from commit a9fe7cfa77ccee64d68732dc3f37d35cbfc27a65)

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 dix/extension.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git xorg-server-1.8.0/dix/extension.c xorg-server-1.8.0/dix/extension.c
index fb83af1..f348665 100644
--- xorg-server-1.8.0/dix/extension.c
+++ xorg-server-1.8.0/dix/extension.c
@@ -83,8 +83,11 @@ AddExtension(char *name, int NumEvents, int NumErrors,
     if (!MainProc || !SwappedMainProc || !MinorOpcodeProc)
         return((ExtensionEntry *) NULL);
     if ((lastEvent + NumEvents > LAST_EVENT) || 
-	        (unsigned)(lastError + NumErrors > LAST_ERROR))
+	        (unsigned)(lastError + NumErrors > LAST_ERROR)) {
+        LogMessage(X_ERROR, "Not enabling extension %s: maximum number of "
+                   "events or errors exceeded.\n", name);
         return((ExtensionEntry *) NULL);
+    }
 
     ext = xalloc(sizeof(ExtensionEntry));
     if (!ext)
-- 
1.7.0.4

